File: Vivino.java
package j.ucdavis22.w5;

/*
 * Jörn Boehnke 2022
 */

import java.time.Duration;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;

import io.github.bonigarcia.wdm.WebDriverManager;

public class Vivino {

    public static void main(String[] args) throws InterruptedException {
        
        WebDriverManager.chromedriver().setup();
        WebDriver driver = new ChromeDriver();
        
        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(20));
        driver.manage().timeouts().scriptTimeout(Duration.ofMinutes(4));
        driver.manage().timeouts().pageLoadTimeout(Duration.ofSeconds(20));
        
        driver.get("https://www.vivino.com/");
        
        Thread.sleep(2_000);
        
        WebElement wines = driver.findElement(By.cssSelector("span[title='Wines']"));
        wines.click();
        
        Thread.sleep(2_000);
        
        WebElement white = driver.findElement(By.xpath("//span[text()='White']"));
        white.click();
        
        Thread.sleep(2_000);
        
        WebElement sparkling = driver.findElement(By.xpath("//span[text()='Sparkling']"));
        sparkling.click();
        
        Thread.sleep(5_000);
        
//        driver.close();
        driver.quit();
        
    }

}

File: Vivino.py
from selenium import webdriver
from selenium.webdriver.common.keys import Keys
import time

driver = webdriver.Chrome(executable_path='/Users/tarang/Downloads/chromedriver')
driver.implicitly_wait(10)
driver.set_script_timeout(120)
driver.set_page_load_timeout(10)

driver.get("https://www.vivino.com/");

#         driver.find_element_by_class_name("className");
#         driver.find_element_by_css_selector".className");
#         driver.find_element_by_id("elementId");
#         driver.find_element_by_link_text("linkText");
#         driver.find_element_by_name("elementName");
#         driver.find_element_by_partial_link_text("partialText");
#         driver.find_element_by_tag_name("elementTagName");
         

wines = driver.find_element_by_css_selector("span[title='Wines']")
wines.click()
time.sleep(2)

white = driver.find_element_by_xpath("//span[text()='White']");
white.click()
time.sleep(2)

sparkling = driver.find_element_by_xpath("//span[text()='Sparkling']");
sparkling.click()
time.sleep(5)

driver.quit()
